airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/helpers/CdcConfigurationHelper.java

```diff
@@ -38,12 +41,35 @@ public class CdcConfigurationHelper {
    * @return list of List<CheckedConsumer<JdbcDatabase, Exception>>
    */
   public static List<CheckedConsumer<JdbcDatabase, Exception>> getCheckOperations() {
-    return List.of(getCheckOperation(LOG_BIN, "ON"),
+    return List.of(getMasterStatusOperation(),
+        getCheckOperation(LOG_BIN, "ON"),
         getCheckOperation(BINLOG_FORMAT, "ROW"),
         getCheckOperation(BINLOG_ROW_IMAGE, "FULL"));
 
   }
 
+  // Checks whether the user has REPLICATION CLIENT privilege needed to query status information about
+  // the binary log files, which are needed for CDC.
+  private static CheckedConsumer<JdbcDatabase, Exception> getMasterStatusOperation() {
+    return database -> {
+      try {
+        database.unsafeResultSetQuery(
+            connection -> connection.createStatement().executeQuery("SHOW MASTER STATUS"),
+            resultSet -> {
+              final String file = resultSet.getString("File");
+              final int position = resultSet.getInt("Position");
+              if (file == null || position == 0) {
```

Comment:
Is it possible that file exists but position is legitimately 0, as in at the top?

Reply:
Yes - actually this is logic from `MySqlTargetPosition`. I'm changing this now to just attempt to execute the query (and not to fail if file or position is empty/null.